xio-core/src/main/java/com/xjeffrose/xio/http/ProxyHandler.java

```diff
@@ -79,8 +82,36 @@ public Request buildRequest(Request request, String proxyHost, String path) {
     return result;
   }
 
+  private Optional<String> extractRemoteAddress(ChannelHandlerContext ctx) {
+    val rawRemoteAddress = ctx.channel().remoteAddress();
+    if (rawRemoteAddress == null) {
+      return Optional.empty();
+    }
+
+    val remoteAddressComponents = rawRemoteAddress.toString().replace("/", "").split(":");
+    if (remoteAddressComponents.length < 1) {
+      return Optional.empty();
+    }
+
+    return Optional.of(remoteAddressComponents[0]);
+  }
+
   private void appendXForwardedFor(ChannelHandlerContext ctx, Request request) {
     // TODO(CK): update request headers
+    val remoteAddress = extractRemoteAddress(ctx);
+    if (remoteAddress.isPresent()) {
+      // if there is rawXFF is valid then return that string with the remote address string
+      // appended,
+      // otherwise return the remote address string
+      val rawXFF = Optional.ofNullable(request.headers().get(X_FORWARDED_FOR));
+      val xForwardedForValue =
+          rawXFF
+              .map(CharSequence::toString)
+              .filter(value -> !value.isEmpty())
+              .map(value -> value + ", " + remoteAddress)
+              .orElse(String.valueOf(remoteAddress));
+      request.headers().set(X_FORWARDED_FOR, xForwardedForValue);
```

Comment:
this should have some tests for various scenarios:
no proxy
multiple proxies
error cases